client/src/pages/VocabDrawer/VocabDrawer.js

```diff
@@ -170,112 +170,118 @@ const VocabDrawer = ({ navigation }) => {
-                image,
-              ),
+            updatedVocabItem = vocabItemResponse.result
+
+            // Push audio recording
+            if (audioRecording && recordingStage === RECORDING.COMPLETE) {
+              const audioResponse = await trackPromise(
+                uploadAudioFile(
+                  currentCourseId,
+                  currentUnitId,
+                  currentLessonId,
+                  updatedVocabItem._id,
+                  audioRecording,
+                ),
+              )
+              updatedVocabItem = audioResponse.result
+            }
+
+            if (image) {
+              const imageResponse = await trackPromise(
+                uploadImageFile(
+                  currentCourseId,
+                  currentUnitId,
+                  currentLessonId,
+                  updatedVocabItem._id,
+                  image,
+                ),
+              )
+              updatedVocabItem = imageResponse.result
+            }
+
+            // Update vocab item in Redux store
+            dispatch(addVocab({ vocab: updatedVocabItem }))
+          } else {
+            const vocabItem = {
+              original: originalText,
+              translation: translatedText,
+              notes: additionalInformation,
+            }
+
+            // Updated vocab item text
+            const vocabItemResponse = await updateVocabItem(
+              currentCourseId,
+              currentLessonId,
+              currentVocabId,
+              vocabItem,
             )
-
-            updatedVocabItem = imageResponse.result
+            updatedVocabItem = vocabItemResponse.result
+
+            // Push audio recording
+            if (audioRecording && recordingStage === RECORDING.COMPLETE) {
+              const audioResponse = await trackPromise(
+                uploadAudioFile(
+                  currentCourseId,
+                  currentUnitId,
+                  currentLessonId,
+                  currentVocabId,
+                  audioRecording,
+                ),
+              )
+
+              updatedVocabItem = audioResponse.result
+            }
+
+            if (image) {
+              const imageResponse = await trackPromise(
+                uploadImageFile(
+                  currentCourseId,
+                  currentUnitId,
+                  currentLessonId,
+                  currentVocabId,
+                  image,
+                ),
+              )
+
+              updatedVocabItem = imageResponse.result
+            }
+            // Update vocab item in Redux store
+            dispatch(updateVocab({ vocab: updatedVocabItem }))
           }
-          // Update vocab item in Redux store
-          dispatch(updateVocab({ vocab: updatedVocabItem }))
-        }
-      },
-      () => {
-        close() // on success, close the modal
-      },
-    )
+        },
+        () => {
+          close() // on success, close the modal
+          setIsDisabled(false)
+        },
```

Comment:
change drawer and add IsDisabled in drawer instead to edit vocab, unit and lesson